config/config_test.go

```diff
@@ -32,18 +33,231 @@ func TestIsStack(t *testing.T) {
+func TestConfigLookup(t *testing.T) {
+	s := sandbox.New(t)
+	s.BuildTree([]string{
+		"d:/dir",
+		"s:/stacks",
+		"s:/stacks/child",
+		"s:/stacks/child/non-stack/stack",
+	})
+
+	cfg := s.Config()
+	node, found := cfg.Lookup("/dir")
+	assert.IsTrue(t, found)
+	assert.IsTrue(t, node.IsEmptyConfig())
+
+	node, found = cfg.Lookup("/stacks")
+	assert.IsTrue(t, found && node.IsStack() && !node.IsEmptyConfig())
+
+	node, found = cfg.Lookup("/stacks/child")
+	assert.IsTrue(t, found && node.IsStack() && !node.IsEmptyConfig())
 
-	res, err := config.IsStack(rootdir, filepath.Join(rootdir, dir))
-	assert.NoError(t, err)
+	node, found = cfg.Lookup("/stacks/child/non-stack")
+	assert.IsTrue(t, found)
+	assert.IsTrue(t, node.IsEmptyConfig())
+
+	node, found = cfg.Lookup("/stacks/child/non-stack/stack")
+	assert.IsTrue(t, found && node.IsStack() && !node.IsEmptyConfig())
+
+	_, found = cfg.Lookup("/non-existent")
+	assert.IsTrue(t, !found)
+
+	stacks := cfg.Stacks()
+	assert.EqualInts(t, 3, len(stacks))
+	assert.EqualStrings(t, "/stacks", project.PrjAbsPath(s.RootDir(), stacks[0].Dir()).String())
+	assert.EqualStrings(t, "/stacks/child", project.PrjAbsPath(s.RootDir(), stacks[1].Dir()).String())
+	assert.EqualStrings(t, "/stacks/child/non-stack/stack", project.PrjAbsPath(s.RootDir(), stacks[2].Dir()).String())
+}
+
+func TestConfigStacksByPaths(t *testing.T) {
+	type testcase struct {
+		name     string
+		layout   []string
+		basedir  string
+		want     []string
+		relpaths []string
+	}
+
+	for _, tc := range []testcase{
+		{
+			name:    "no stacks, no relpaths",
+			layout:  []string{},
+			basedir: "/",
+		},
+		{
+			name:    "no stacks, absolute path",
+			layout:  []string{},
+			basedir: "/",
+			relpaths: []string{
+				"/",
+			},
+		},
+		{
+			name:    "no stacks, rel path",
+			layout:  []string{},
+			basedir: "/",
+			relpaths: []string{
+				"/",
+			},
+		},
+		{
+			name: "single stack, no matching path",
+			layout: []string{
+				"s:stack",
+				"d:some/place",
+			},
+			basedir: "/some/place",
+			relpaths: []string{
+				"/test",
+				"../../test",
+			},
+		},
+		{
+			name: "single stack, 1 matching absolute path",
+			layout: []string{
+				"s:stack",
+				"d:some/place",
+			},
+			basedir: "/some/place",
+			relpaths: []string{
+				"/test",
+				"../../test",
+				"/stack",
+			},
+			want: []string{
+				"/stack",
+			},
+		},
+		{
+			name: "single stack, 1 matching relpath",
+			layout: []string{
+				"s:stack",
+				"d:some/place",
+			},
+			basedir: "/some/place",
+			relpaths: []string{
+				"/test",
+				"../../test",
+				"../../stack",
+			},
+			want: []string{
+				"/stack",
+			},
+		},
+		{
+			name: "single stack, multiple match returns once",
+			layout: []string{
+				"s:stack",
+				"d:some/place",
+			},
+			basedir: "/some/place",
+			relpaths: []string{
+				"../../stack",
+				"../../stack",
+			},
+			want: []string{
+				"/stack",
+			},
+		},
+		{
+			name: "single stack, 1 matching relpath",
```

Comment:
Isn't this test a duplicate from [here](https://github.com/mineiros-io/terramate/pull/631/files#diff-373f5578b8f362a364b50b0d01919a57da64c128547ca10d49df8ae422237c82R139) ?